pkg/endpoint/restore.go

```diff
@@ -298,10 +298,13 @@ func (e *Endpoint) restoreIdentity() error {
 			return ErrNotAlive
 		case <-gotInitialGlobalIdentities:
 		}
+	}
```

Comment:
Second patch LGTM.

I'm curious, do you have an easy way to reproduce these connectivity issues during agent restart? We have some tests like the Upgrade tests that attempt to do this, but it sounds like your testing is able to more reliably pick up on issues like this. If we can integrate such tests into the Cilium CI, we can hopefully catch these bugs in future before they get merged into the tree.

Reply:
I see there was already discussion on this above in the PR, I missed that initially :)

We do not need to resolve this testing question to merge the fix, but it would be a nice follow-up to prevent future regressions.